source/howto/AddUsersOpenStackVMInstance.rst

```diff
@@ -0,0 +1,148 @@
+
+                Location: r89.harwell.europe hpd r89rack241
+                  Branch: xfu59478/mr_build_centos_aq_image (sandbox)
+              Archetype: cloud
+            Personality: nubesvms
+        Operating System: centos7x-x86_64
+          Snapshot Date: 2019-05-20
+
+#. Login as Root
+
+    Use the `sudo` command to login in as root:
+
+    .. code-block:: shell
+
+      [FEDID@preprod-horizon1 ~]$ sudo su -
+
+      [root@preprod-horizon1 ~]#
+
+#. Adding a New User to the System using the script (addusers.sh) below. Edit the script and save it.
+
+    .. code-block:: shell
+
+      declare -A KEYS
+
+      KEYS[ntint]="ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDOXk3OukWtXQHbyhxuG2ST/zhGfoPHQi/5Zvt5PELskztdPMZRKn3bCqSk0nZQppoFgGsze93CWcfiY2ppUTGoKbZMnmDSjfO0DUFXB/ADNJapo53Yd0lRnxM45JuFT+AS8cg/TalzbBeXcO2Tyr/hFTm2p7b5PQZgny9RwdXI6W0rTyxnktBeMoy6bMepPueZgUhJVJIKFaAymvVlmeZHhlRX2MB/Y7OTpETNiuTbVaSw3kI57rZ16pgvCWwl0RyQon9V6yBsduaF9MAHTVlsuKBd7S7Q1V/19CpaAKdowq+HrZ5lbu3H7+GvX8LSpNAGzOtuNOxm50U/jcOWaUHxMtNCTMm1oHtdN13vADtTpU5MbtzJ89q/pSTfJVbX0oj8AFN+gtb4hlvKWaWwEUwOe5P4JlVIX4PjHZKzAdf+s1cumJ9sbjgvFIKzrGes6lpUBdjw0WGQnB0GLSkml7Hu6x8jecd4hC04tSqgJLw+KXZysVYCUkIuDEp/GKGv8aM= ntint@byod-vpn-168-0067.rl.ac.uk"
+
+      KEYS[mtint]="ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDOXk3OukWtXQHbyhxuG2ST/zhGfoPHQi/5Zvt5PELskztdPMZRKn3bCqSk0nZQppoFgGsze93CWcfiY2ppUTGoKbZMnmDSjfO0DUFXB/ADNJapo53Yd0lRnxM45JuFT+AS8cg/TalzbBeXcO2Tyr/hFTm2p7b5PQZgny9RwdXI6W0rTyxnktBeMoy6bMepPueZgUhJVJIKFaAymvVlmeZHhlRX2MB/Y7OTpETNiuTbVaSw3kI57rZ16pgvCWwl0RyQon9V6yBsduaF9MAHTVlsuKBd7S7Q1V/19CpaAKdowq+HrZ5lbu3H7+GvX8LSpNAGzOtuNOxm50U/jcOWaUHxMtNCTMm1oHtdN13vADtTpU5MbtzJ89q/pSTfJVbX0oj8AFN+gtb4hlvKWaWwEUwOe5P4JlVIX4PjHZKzAdf+s1cumJ9sbjgvFIKzrGes6lpUBdjw0WGQnB0GLSkml7Hu6x8jecd4hC04tSqgJLw+KXZysVYCUkIuDEp/GKGv8aM= mtint@byod-vpn-168-0067.rl.ac.uk"
+
+      groupadd sshusers
+
+      for FEDID in "${!KEYS[@]}"
+        do
+          SSH_PUBLIC_KEY=${KEYS[$FEDID]}
+          echo "Creating USER $FEDID"
+          echo "Adding KEY $SSH_PUBLIC_KEY"
+          echo " "
+          echo -n "Add $FEDID to sudo group (y/n)? "
+          read answer
+          if [ "$answer" != "${answer#[Yy]}" ] ;then
+            useradd $FEDID -g wheel;
+            echo "$FEDID ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers;
+          else
+          useradd $FEDID;
+          fi
+          usermod -a -G sshusers $FEDID
+          mkdir /home/$FEDID/.ssh
+          echo "$SSH_PUBLIC_KEY" >> /home/$FEDID/.ssh/authorized_keys
+          chown -R $FEDID /home/$FEDID/.ssh
+          echo "User USER $FEDID Created"
+          echo " "
+      done
+          chmod 0440 /etc/sudoers
+
+This script should only be run once. 
+
+User (FEDID) and SSH KEYS are stored in an Array with the format KEY[FEDID]="SSH KEY"
+
+.. code-block:: shell
+
+  KEYS[cog12345]="ssh-rsa xxxxx"
+
+After editing the script with users and keys run by typing
+
+.. code-block:: shell
+  
+  source addusers.sh
```

Comment:
This should be `sh addusers.sh` not source, as it can modify their terminal env using source